src/selectors/getSelectedFrameLocation.js

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+import { getSelectedLocation } from "../reducers/sources";
+import { getSelectedFrame } from "../reducers/pause";
+import { isOriginalId } from "devtools-source-map";
+
+export default function getSelectedFrameLocation(state: OuterState) {
+  const selectedLocation = getSelectedLocation(state);
+  const isGeneratedSource = !isOriginalId(selectedLocation.sourceId);
+  const selectedFrame = getSelectedFrame(state);
+
+  if (!selectedFrame) {
+    return;
+  }
+
+  return isGeneratedSource
+    ? selectedFrame.generatedLocation || selectedFrame.location
```

Comment:
are there cases when there won't be `selectedFrame.generatedLocation`?

Reply:
I did not encountered such use case. I just used the same logic we have wrote in getCallStackFrames and also saw in visibleBreakpoint , assuming there could be such use case. 
I can remove it if you feel comfortable with it.